test/e2e/common/util.go

```diff
@@ -184,3 +186,63 @@ func RestartNodes(c clientset.Interface, nodes []v1.Node) error {
 	}
 	return nil
 }
+
+func RecreateNodes(c clientset.Interface, nodes []v1.Node) error {
```

Comment:
The test is skipped unless the provider is gke/gce but without looking at the exact implementation I wouldn't know that this method is also gki/gce specific. Putting it ito the `test/e2e/common/util.go` seems like it is overly specific an implementation to belong here.

`./test/e2e/framework/providers/gce` has a few other gce specific tests and seems like an intuitive place. Is there a reason it didnt go there? It also seems to have other helpers that might be useful.

Reply:
Thanks for letting me know. This is only run for gke/gce since it uses the `gcloud` command, like RestartNodes. Would it be more suitable that both of these functions go into `./test/e2e/framework/providers/gce`?

Reply:
Yes it does seem to me that it would be the natural place to put it.

And if this recreation of nodes is going to be a generic lifecycle method, then it seems like it would make sense to change the `ProviderInterface` in order to add the `RecreateNodes` method.  I'm not too familiar with these features though but for example I see `ResizeGroup` as part of that interface and defined for each provider.

Putting it there you would just want to review the other contracts a bit more and make sure you are being consistent with the type of parameters you're requiring, values you're returning, and types of actions you take within the method.

Reply:
Thanks. I moved the test, since there isn't anything for recreating nodes in the `ProviderInterface` yet.

Reply:
Moved the test to the gce provider test package and exported some utilities functions to `test/e2e/framework/util` from `test/e2e/lifecycle/restart`.